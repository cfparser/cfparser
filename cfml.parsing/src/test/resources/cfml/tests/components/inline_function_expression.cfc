component {

	function dotrim(data, callback){
		callback(trim(data));
	}

	function bar(){
		dotrim("hello world", function(result){

		});
	}

}
