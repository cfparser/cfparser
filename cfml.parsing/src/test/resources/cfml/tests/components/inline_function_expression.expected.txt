/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
FUNCTION                 <function>
IDENTIFIER               <dotrim>
'('                      <(>
IDENTIFIER               <data>
','                      <,>
IDENTIFIER               <callback>
')'                      <)>
'{'                      <{>
IDENTIFIER               <callback>
'('                      <(>
IDENTIFIER               <trim>
'('                      <(>
IDENTIFIER               <data>
')'                      <)>
')'                      <)>
';'                      <;>
'}'                      <}>
FUNCTION                 <function>
IDENTIFIER               <bar>
'('                      <(>
')'                      <)>
'{'                      <{>
IDENTIFIER               <dotrim>
'('                      <(>
OPEN_STRING              <">
STRING_LITERAL           <hello world>
CLOSE_STRING             <">
','                      <,>
FUNCTION                 <function>
'('                      <(>
IDENTIFIER               <result>
')'                      <)>
'{'                      <{>
'}'                      <}>
')'                      <)>
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          function
          (identifier dotrim)
          (
          (parameterList
            (parameter (identifier data))
            ,
            (parameter (identifier callback))
          )
          )
          (compoundStatement
            {
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall
                          (identifierOrReservedWord (identifier callback))
                          (
                          (argumentList
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (unaryExpression
                                      (memberExpression
                                        (functionCall
                                          (identifierOrReservedWord (identifier trim))
                                          (
                                          (argumentList
                                            (argument
                                              (startExpression
                                                (compareExpression
                                                  (baseExpression (unaryExpression (memberExpression (identifier data))))
                                                )
                                              )
                                            )
                                          )
                                          )
                                        )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      (element
        (functionDeclaration
          function
          (identifier bar)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (startExpression
                (compareExpression
                  (baseExpression
                    (unaryExpression
                      (memberExpression
                        (functionCall
                          (identifierOrReservedWord (identifier dotrim))
                          (
                          (argumentList
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (unaryExpression
                                      (primaryExpression
                                        (literalExpression (stringLiteral " (stringLiteralPart hello world) "))
                                      )
                                    )
                                  )
                                )
                              )
                            )
                            ,
                            (argument
                              (startExpression
                                (compareExpression
                                  (baseExpression
                                    (anonymousFunctionDeclaration
                                      function
                                      (
                                      (parameterList (parameter (identifier result)))
                                      )
                                      (compoundStatement { })
                                    )
                                  )
                                )
                              )
                            )
                          )
                          )
                        )
                      )
                    )
                  )
                )
              )
              ;
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/